app/controllers/etds_controller.rb

```diff
@@ -21,14 +21,21 @@ def index
 
   # POST /etds
   # receives xml data from peoplesoft
-  def create
+  def create # rubocop:disable Metrics/AbcSize
     logger.info("RAW XML: #{request.raw_post}")
 
     return unless submission
 
-    @submission.update!(submission_attributes)
-    peoplesoft_actions
-    render_created
+    Submission.transaction do
+      submission.update!(submission_attributes)
+
+      peoplesoft_service.new_reader_action(readers:, reader_action_attributes:)
+      peoplesoft_service.new_registrar_action(registrar_action_attributes:)
+
+      submission.generate_and_attach_augmented_file!
```

Comment:
This is the sigpage-related change.

Note, per the old ETD app, we want to generate/regenerate the augmented PDF in a few different scenarios:

1. When the registrar POSTs to `/etds` (The change on line 35 fits this scenario.)
2. When the student uploads their dissertation file
3. When the student submits to the registrar

We do this to make sure we don't have an augmented PDF with outdated info when it's submitted. I could be convinced heracles can use a more simplified set of assumptions, but I thought it wise to err on the side of caution.

Reply:
I think your assumptions here are correct, I tend to feel like knowingly swallowing an error here is a bit of smell (though could be swayed otherwise). Would you consider changing line 122 to something like `@new_submission = submission.new_record` and then you could do something like:

```
return render status: :created, plain: "#{submission.driud} created" if @new_submission

submission.generate_and_attach_augmented_file!
render status: :ok, plain: "#{submission.driud} updated"
```

This would allow us to render the proper 201 on create but 200 on update, as well as avoid swallowing errors that may be indicators of legitimate problems.

Mostly thinking out loud here, and happy to discuss.

Reply:
@aaron-collier I like the idea of being a better web citizen, and returning more appropriate HTTP statuses to UIT (even if they don't care or use them). I think we would still need to swallow the error because it's possible, though not likely, we get multiple POSTs for an ETD before the dissertation file is uploaded.

Reply:
I added the HTTP status switching behavior for now